web3/conversions.nim

```diff
@@ -126,112 +148,137 @@ template wrapValueError(body: untyped) =
   except ValueError as exc:
     r.raiseUnexpectedValue(exc.msg)
 
-proc writeValue*(w: var JsonWriter[JrpcConv], val: UInt256)
-      {.gcsafe, raises: [IOError].} =
-  w.writeValue("0x" & val.toHex)
-
-# allows UInt256 to be passed as a json string
-proc readValue*(r: var JsonReader[JrpcConv], val: var UInt256)
-      {.gcsafe, raises: [IOError, JsonReaderError].} =
-  # expects base 16 string, starting with "0x"
-  let tok = r.tokKind
-  if tok != JsonValueKind.String:
-    r.raiseUnexpectedValue("Expected string for UInt256, got=" & $tok)
-  let hexStr = r.parseString()
-  if hexStr.len > 64 + 2: # including "0x"
-    r.raiseUnexpectedValue("String value too long for UInt256: " & $hexStr.len)
-  if hexStr.invalidQuantityPrefix:
-    r.raiseUnexpectedValue("UInt256 value has invalid leading 0")
-  wrapValueError:
-    val = hexStr.parse(StUint[256], 16)
+func valid(hex: string): bool =
+  for x in hex:
+    if x notin HexDigits: return false
+  true
 
-proc writeHexValue(w: var JsonWriter[JrpcConv], v: openArray[byte])
+proc writeHexValue(w: var JsonWriter, v: openArray[byte])
       {.gcsafe, raises: [IOError].} =
   w.stream.write "\"0x"
   w.stream.writeHex v
   w.stream.write "\""
 
-proc writeValue*(w: var JsonWriter[JrpcConv], v: DynamicBytes)
-      {.gcsafe, raises: [IOError].} =
-  writeHexValue w, distinctBase(v)
+#------------------------------------------------------------------------------
+# Well, both rpc and chronicles share the same encoding of these types
+#------------------------------------------------------------------------------
```

Comment:
`type JsonFlavors = JrpcConv | DefaultFavlor`?

Reply:
Yup, thanks. That will reduce maintenance cost.